README.md

```diff
@@ -1,38 +1,85 @@
-# Initial deploy
+# Radio Bern RaBe Icecast web theme
+[Icecast](http://icecast.org/) streaming server web layout (CSS, HTML and
+images) for [Radio Bern RaBe](http://www.rabe.ch) based on the original Icecast
+layout.
 
-Login to stream.rabe.ch as root.
 
-	cd ~
-	cd /usr/share/
+## Installation via RPM on CentOS
+The RaBe Icecast web theme is packaged for CentOS 7 and available on [Radio
+RaBe's Audio Packages for Enterprise Linux
+repository](https://build.opensuse.org/project/show/home:radiorabe:audio). It
+can be installed as follows:
 
-Backup original configuration:
+```bash
+# Install the RaBe APEL repository configuration
+curl -o /etc/yum.repos.d/home:radiorabe:audio.repo \
+     http://download.opensuse.org/repositories/home:/radiorabe:/audio/CentOS_7/home:radiorabe:audio.repo
 
-	mv icecast icecast.orig
+# Install the package
+yum install rabe-icecast-theme
+```
 
-Install templates from the git repository:
+[Configure the Icecast server](#icecast-configuration) to use the new theme.
 
-	git clone https://github.com/radiorabe/rabe-icecast-theme.git
-	mv ~/rabe-icecast-theme /usr/share/icecast
 
-Restore authentication configs from original files:
+## Manual installation
+For all other distributions you can manually install the RaBe Icecast web theme
+via Git (make sure you have `git` installed):
 
-	mv icecast.orig/auth/ icecast/
+```bash
+# Install the web theme from the git repository:
+cd /usr/local/share
+git clone https://github.com/radiorabe/rabe-icecast-theme.git
 
-Fix permissions for all files:
+# To update the web theme
+cd /usr/local/share/rabe-icecast-theme
+git pull
+```
 
-	find /usr/share/icecast -type d -exec chmod 755 {} \;
-	find /usr/share/icecast -type f -exec chmod 644 {} \;
+[Configure the Icecast server](#icecast-configuration) to use the new theme.
 
-# Update templates from repository
+## Icecast configuration
+The Icecast server must be configured to use the newly installed web theme.
+Edit the Icecast configuration XML and point the `webroot` and `adminroot` to
+the new theme directories:
+```bash
+vi /etc/icecast.xml
+```
 
-Login to stream.rabe.ch as root:
+```xml
+<icecast>
```

Comment:
Have we considered automating this in a fashion similar to what the rabe-motd package does?

Reply:
Yes, but as part of post-installation steps (i.e. with the help of a configuration management system) and not from the RPM itself. It is a fundamental bad design practice to let a package modify or move (configuration) files owned by a different package. Although it's possible with RPM, most other packaging system would prevent this for good reasons.

Reply:
> **most** other packaging systems

That sample size is just silly...

Reply:
> most other packaging systems
> 
> That sample size is just silly...

sample size?

Reply:
I don't understand what **most** other packaging systems you are alluding to have to do with this this. We aren't packaging for or deploying on those.

I don't think this belongs into a configuration-management system. It doesn't need changes in day 2 ops and it's pretty much safe to assume that folks who install this want it per default.

RPM provides heaps of triggers for solving exactly this kind of problem and given that our config management isn't ready (and won't be managing icecast anytime soon) it would really make life easier if this could be automated to somewhat ensure consistence on all our icecast deploys.

IBM developerWorks [has a nice article](https://www.ibm.com/developerworks/library/l-rpm2/) explaining all the triggers and it doesn't look like there is something missing needed to automate this. Doing this in rpm won't conflict with later use of config management, modern frameworks can use tooling like augeas to manage individual nodes of an XML document without the need for tempting.

Reply:
> I don't understand what most other packaging systems you are alluding to have to do with this this. We aren't packaging for or deploying on those.

That's correct we don't. Just because it's possible doesn't mean, that it's a good idea - and that's where other package systems seems to agree, that's all I've tried to say her. Configuration and inter-package file modification protection is there for a reason.

In RPM you have the `%config(noreplace)` macro which prevents the rpm command from overwriting a (modified) file. The icecast Spec file is exactly doing that for the `icecast.xml`, that request would be effectively ignored from this third-party package.


> RPM provides heaps of triggers for solving exactly this kind of problem

while the triggers are really nice and flexible, I still strongly disagree that one should use them to modify a protected configuration file originating from an other package. Dropping a configuration snipped into a *configuration.d* style directory, would be the way to go - unfortunately icecast doesn't seems to support that.

> and given that our config management isn't ready (and won't be managing icecast anytime soon) it would really make life easier if this could be automated to somewhat ensure consistence on all our icecast deploys.

Absolutely. As of this, a Kickstart snippet is currently taking care of that.